eagleEvents/seating_chart_ga.py

```diff
@@ -7,182 +7,346 @@
+        logbook.chapters["like"].header = "avg", "std", "min", "max"
+
+    # Create seed
+    seed = []
+    for i in range(50):
+        heuristic = []
+        for guest_number in random.sample(guest_numbers, len(guest_numbers)):
+            if guest_number == Table.EMPTY_SEAT:
+                heuristic.append(guest_number)
+                continue
+            if guest_number in heuristic:
+                continue
+            heuristic.append(guest_number)
+            pref_dict = guest_lookup[guest_number]
+            if pref_dict is None:
+                continue
+            for other_guest_num, seating_pref_value in pref_dict.items():
+                if seating_pref_value == SeatingPreference.LIKE.value and not (other_guest_num in heuristic):
+                    heuristic.append(other_guest_num)
+        heuristic += [Table.EMPTY_SEAT for _ in range(num_extra_seats)]
+        seed.append(heuristic)
+
+    toolbox.register("indices", random.sample, table_assignments, len(table_assignments))
+    toolbox.register("individual", tools.initIterate, creator.Individual, toolbox.indices)
+    toolbox.register("population", tools.initRepeat, list, toolbox.individual)
+    toolbox.register("population_guess", init_population, list, creator.Individual, seed)
+
+    pop = toolbox.population_guess(n=NIND, pct_heuristic=INIT_PCT_GUESS)
+
+    # Evaluate the first generation
+    update_fitnesses(lookup_and_size, pop)
+    if collect_stats:
+        update_stats(mstats, logbook, 0, pop)
+
+    generation_number = 1
+    while not should_terminate(hall_of_fame, generation_number, total_like_preferences, NGEN):
+        pop[:] = do_generation(hall_of_fame, lookup_and_size, pop, NIND)
+        if collect_stats:
+            update_stats(mstats, logbook, generation_number, pop)
+        pop.extend(hall_of_fame)
+        hall_of_fame = sorted(pop, key=lambda indiv: indiv.fitness.values[1] - (indiv.fitness.values[0] * 2),
+                              reverse=True)[:HALL_OF_FAME_SIZE]
+        if log_output:
+            print("Generation: {}".format(generation_number))
+            print("Current Best: {a} dislikes, {b} likes".format(a=hall_of_fame[0].fitness.values[0],
+                                                                 b=hall_of_fame[0].fitness.values[1]))
+        generation_number += 1
+
+    winner = hall_of_fame[0]
+    tables = []
+    for t in range(num_tables):
+        table = Table(event)
+        table.number = t + 1
+        guest_numbers_at_table = winner[t * event.table_size.size: (t + 1) * event.table_size.size]
+        table.guests = list(filter(lambda g: g.number in guest_numbers_at_table, event._guests))
+        tables.append(table)
+    if collect_stats:
+        return tables, logbook, winner, total_like_preferences, total_dislike_preferences
+    else:
+        return tables
+
+
+def guest_list_to_nested_dict(guests):
+    guests_dict = dict()
+    for guest in guests:
```

Comment:
For larger guest lists, it may be worthwhile it to parallelize this

Reply:
Not worth it, only done once